game/core/src/main/java/com/gg/game/model/math/Shape.java

```diff
@@ -0,0 +1,12 @@
+package com.gg.game.model.math;
+
+import com.gg.game.model.utils.Point;
+import com.gg.game.model.utils.Vector;
+
+public interface Shape {
+    boolean contains(float x, float y);
+
+    boolean contains(Point point);
+
+    boolean contains(Vector point);
```

Comment:
See comment on circle. This interface should probably have a intersects-method.

Reply:
I do not know of a good way to make the interface method declaration accept any form of implementation of the interface as arguments.

Something like 
```suggestion
    boolean intersects(ShapeImpl shapeImpl implements Shape);
```
is not possible from what I know.

Do you know of any way to do it?